### PR TITLE
Impl pretty-printing for AugmentedScriptSet

### PR DESCRIPTION
**Pull request by crlf0710**

### CODE REVIEW

File: src/mixed_script.rs [hunk 1]
```diff
@@ -67,14 +65,50 @@ impl From<&'_ str> for AugmentedScriptSet {
...
             jpan: true,
             kore: true,
         }
     }
 }
 
+impl Debug for AugmentedScriptSet {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if self.is_empty() {
+            write!(f, "AugmentedScriptSet{{∅}}")?;
```
Comment (Manishearth) [inline src/mixed_script.rs]: Can we have a space after the name?

File: src/mixed_script.rs [hunk 2]
```diff
@@ -67,14 +65,50 @@ impl From<&'_ str> for AugmentedScriptSet {
...
+        } else if self.is_all() {
+            write!(f, "AugmentedScriptSet{{ALL}}")?;
+        } else {
+            write!(f, "AugmentedScriptSet{{")?;
+            let mut first_entry = true;
+            let hanb = if self.hanb { Some("Hanb") } else { None };
+            let jpan = if self.jpan { Some("Jpan") } else { None };
+            let kore = if self.kore { Some("Kore") } else { None };
+            for writing_system in None.into_iter().chain(hanb).chain(jpan).chain(kore) {
```
Comment (Manishearth) [inline src/mixed_script.rs]: we can just chain an `base.iter.map(Script::short_name)` here

File: src/mixed_script.rs [hunk 3]
```diff
@@ -67,14 +65,50 @@ impl From<&'_ str> for AugmentedScriptSet {
...
+                    write!(f, ", ")?;
+                } else {
+                    first_entry = false;
+                }
+                write!(f, "{}", script.short_name())?;
+            }
+            //for script in self.hanb
+            write!(f, "}}")?;
+        }
+        //if self.base.is_common() || self.base.is_
```
Comment (Manishearth) [inline src/mixed_script.rs]: Remove the comments

File: src/restriction_level.rs [hunk 4]
```diff
@@ -58,13 +58,21 @@ impl RestrictionLevelDetection for &'_ str {
             }
         }
 
+        fn single_item<I: Iterator>(mut iter: I) -> Option<I::Item> {
```
Comment (Manishearth) [inline src/restriction_level.rs]: Can we instead implement this via `count_ones()`?

File: src/mixed_script.rs [hunk 5]
```diff
@@ -67,14 +65,50 @@ impl From<&'_ str> for AugmentedScriptSet {
 impl Default for AugmentedScriptSet {
     fn default() -> Self {
         AugmentedScriptSet {
-            base: ScriptExtension::Single(Script::Common),
+            base: ScriptExtension::from(Script::Common),
```
Comment (Manishearth) [inline src/mixed_script.rs]: nit: I prefer `.into()`

File: src/restriction_level.rs (hunk 4 above)
Comment (crlf0710) [reply src/restriction_level.rs]: Do you mean accessing the bitset underlying `ScriptExtension` directly? We cannot access the implementation detail from this crate...

File: src/mixed_script.rs (hunk 3 above)
Comment (crlf0710) [reply src/mixed_script.rs]: oops

File: src/mixed_script.rs (hunk 2 above)
Comment (crlf0710) [reply src/mixed_script.rs]: Indeed! Thanks

File: src/mixed_script.rs (hunk 1 above)
Comment (crlf0710) [reply src/mixed_script.rs]: Sure

File: src/restriction_level.rs (hunk 4 above)
Comment (Manishearth) [reply src/restriction_level.rs]: Right, add that to the original crate. Maybe a `len()` method

Comment (crlf0710) [reply src/restriction_level.rs]: https://github.com/unicode-rs/unicode-script/pull/4/

Comment (crlf0710) [reply src/restriction_level.rs]: The PR is merged, now we need a new release of that crate.

Comment (Manishearth) [reply src/restriction_level.rs]: Done.

**Comment by crlf0710** [comment]
Rebased and addressed the review comments.

**Pull request merged**